### ISSUE TITLE
Relation Widget Displays Value Instead of DisplayValue on Selection

### ISSUE THREAD

**Issue by Undistraction** [netlify/netlify-cms#1982]
**Describe the bug**
When both a `displayFields` and `valueFields` are supplied to a `relations` widget, the widget shows the correct values in the autosuggest, but as soon as a value is selected, displays  the `valueField` value.

**To Reproduce**
Configure a `tip` collection:
```
- name: tip
  label: Tips
  folder: src/content/tips
  create: true
  fields:
    - name: uuid
      label: UUID
      widget: uuid
    - name: name
       label: Name
       widget: string
    - name: body
       label: Body
       widget: markdown
```

Configure a relation widget with the following:

```
- name: tip
  label: Tip
  widget: relation
  collection: tip
  searchFields: ['name', 'body']
  valueField: uuid
  displayFields: ['name']
```

When typing text into the widget, the `name` field of the tips are used. However once a selection is made, the name is no longer displayed, and is replaced by that tip's `uuid` field.

**Expected behavior**
The displayed value should always be the value of the `displayFields`, i.e. the tip's `name` field.

**Screenshots**

Suggestions
<img width="836" alt="screenshot 2018-12-31 at 17 31 22" src="https://user-images.githubusercontent.com/161489/50564988-273eff80-0d22-11e9-8662-58e3f6ab7a2a.png">

Selection
<img width="836" alt="screenshot 2018-12-31 at 17 33 31" src="https://user-images.githubusercontent.com/161489/50564999-3faf1a00-0d22-11e9-91fe-78f5ebc86d3c.png">

**Applicable Versions:**
 - Netlify CMS version: netlify-cms 2.3.2
 - Git provider: Gitlab
 - OS: OSX
 - Browser version: 71.0.3578.98
 - Node.JS version: 8.11.3

**Comment by Undistraction** [comment]
@papandreou Thanks. I misread that documentation, but this definitely _feels_ like a bug. It makes no sense to me that the value you have selected should be replaced by a completely unrelated value. Could be add another param to control the rendering of the selected value in the same way you can format the display of a date, but still save the raw date to the field?

**Comment by erquhart** [comment]
Yeah...I can't think of a reason why it should be this way. Definitely open to a PR for this.

**Comment by erquhart** [comment]
#1936 resolves this, and is very close to merge ready. If you're up for reviewing that would help!

**Comment by barthc** [comment]
Issue fixed in #1936
